src/status_im/contexts/settings/wallet/effects.cljs

```diff
@@ -0,0 +1,45 @@
+(ns status-im.contexts.settings.wallet.effects
+  (:require [native-module.core :as native-module]
+            [promesa.core :as promesa]
+            [status-im.contexts.syncing.events :as syncing-events]
+            [status-im.contexts.syncing.utils :as sync-utils]
+            [utils.re-frame :as rf]
+            [utils.security.core :as security]
+            [utils.transforms :as transforms]))
+
+(rf/reg-fx :effects.connection-string/export-keypair
+ (fn [{:keys [key-uid sha3-pwd keypair-key-uid on-success on-fail]}]
+   (let [config-map (transforms/clj->json {:senderConfig {:loggedInKeyUid   key-uid
+                                                          :keystorePath     ""
+                                                          :keypairsToExport [keypair-key-uid]
+                                                          :password         (security/safe-unmask-data
+                                                                             sha3-pwd)}
+                                           :serverConfig {:timeout 0}})]
+     (-> (native-module/get-connection-string-for-exporting-keypairs-keystores
+          config-map)
+         (promesa/then (fn [response]
+                         (if (sync-utils/valid-connection-string? response)
+                           (on-success response)
+                           (on-fail (throw (js/Error.
+                                            "generic-error: failed to get connection string"))))))
+         (promesa/catch on-fail)))))
+
+(rf/reg-fx :effects.connection-string/import-keypair
+ (fn [{:keys [key-uid sha3-pwd keypairs-key-uids connection-string on-success on-fail]}]
+   (let [config-map (.stringify js/JSON
+                                (clj->js
+                                 {:receiverConfig
+                                  {:loggedInKeyUid   key-uid
+                                   :keystorePath     ""
+                                   :password         (security/safe-unmask-data
+                                                      sha3-pwd)
+                                   :keypairsToImport keypairs-key-uids}}))]
+     (-> (native-module/input-connection-string-for-importing-keypairs-keystores
+          connection-string
+          config-map)
+         (promesa/then (fn [res]
+                         (if-let [error (when (syncing-events/extract-error res)
```

Comment:
if I'm not mistaken, this error extraction is already part of `promesa`:

https://github.com/status-im/status-mobile/blob/841a6140e339d878aa7a0bec3acebdc55bb1b04e/src/native_module/utils.cljs#L7-L23

We can skip this whole check

Reply:
cc @clauxx

Reply:
@smohamedjavid as i checked if it throws error it still trigger `promesa/then` and we should still check for error.